src/os_win32.c

```diff
@@ -3133,7 +3142,14 @@ mch_dirname(
 	return FAIL;
     }
 #endif
-    return (GetCurrentDirectory(len, (LPSTR)buf) != 0 ? OK : FAIL);
+    if (GetCurrentDirectory(len, (LPSTR)buf) == 0)
+	return FAIL;
+
+    if (GetLongPathNameA((LPSTR)buf, (LPSTR)abuf, _MAX_PATH) == 0)
```

Comment:
I think `len` should be used instead of `_MAX_PATH`.

Reply:
The size of ```abuf``` is ```_MAX_PATH + 1```, so there it is.